landa/water_body_management/change_log.py

```diff
@@ -0,0 +1,200 @@
+import json
+
+import frappe
+from frappe.query_builder.custom import ConstantColumn
+from frappe.utils import cint
+
+
+class ChangeLog:
+	def get_logs(self, from_datetime: str):
+		"""
+		Get queried and formatted change logs.
+		"""
+		self.changed_data = self._get_changed_data(from_datetime) or []
+		return self._get_formatted_changes()
+
+	def _get_changed_data(self, from_datetime: str):
+		"""
+		Get created, modified and deleted Water Body, Fish Species, WBMLO, Attached File data.
+		"""
+		created_modified = self._get_version_log_query(from_datetime)
+		deleted = self._get_deleted_document_query(from_datetime)
+		union_query = created_modified.union(deleted)
+		return (frappe.qb.from_(union_query).select(union_query.star).orderby(union_query.creation)).run(
+			as_dict=True
+		)
+
+	def _get_formatted_changes(self):
+		"""
+		Format changes dict to be displayed in the Change Log API.
+		"""
+		changes = []
+		for entry in self.changed_data:
+			change_log = self.format_change(entry)
+			if change_log:
+				changes.append(change_log)
+
+		return changes
+
+	def format_change(self, entry: dict):
+		changed_data = frappe._dict(json.loads(entry.data))
+		event = self._get_event(entry, changed_data)
+
+		# Modified dependencies (File and WBMLO)
+		if entry.doctype in ("File", "Water Body Management Local Organization"):
+			change_log = self._build_dependency_change_log(entry, changed_data)
+			if change_log:
+				return change_log
+
+		# Created/Deleted Water Body/Fish Species
+		if event in ("Created", "Deleted"):
+			return {
+				"id": entry.name,
+				"doctype": entry.doctype,
+				"docname": entry.docname,
+				"event": event,
+				"datetime": entry.creation,
+			}
+
+		# Modified Water Body/Fish Species Log
+		change_log = self._build_modified_change_log(entry, changed_data, event)
+		if change_log:
+			return change_log
+
+		return None
```

Comment:
```suggestion
		return self._build_modified_change_log(entry, changed_data, event)
```

This should be equivalent, but shorter